core/lib/dal/migrations/20240111084243_events-without-eth-transfer-indexes.up.sql

```diff
@@ -0,0 +1,4 @@
+ALTER TABLE events
+    ADD COLUMN IF NOT EXISTS event_index_in_block_without_eth_transfer INT DEFAULT NULL;
+ALTER TABLE events
+    ADD COLUMN IF NOT EXISTS event_index_in_tx_without_eth_transfer INT DEFAULT NULL;
```

Comment:
According to quick research

> In Postgres 10 or older (such as OP's 9.6), any DEFAULT clause even DEFAULT NULL will require a full rewrite of the table, which might take essentially forever for large tables[1].

so this is not safe. We can try it on stage2 (although please copy a subset of table first, don't amend `eevents` table there) , but I think it won't work. 

We may find a better migration approach (just remove `DEFAULT`?) but even then we might negatively impact insert time to this table which will result in lower TPS. Is there an alternative solution?

Reply:
We don't use postgres 10 or older, don't we? I thought we're at 14 or at least 12?

Reply:
![image](https://github.com/matter-labs/zksync-era/assets/12111581/4c42b63f-9d59-4bca-b460-46e96989ebe5)

Reply:
E.g. from [the same response](https://stackoverflow.com/questions/47709593/postgresql-add-column-default-null-locks-and-performance) (I guess) below:

> This was changed with Postgres 11, Vao Tsun's answer is only correct for these, [per the Postgres 11 documentation](https://www.postgresql.org/docs/11/sql-altertable.html#SQL-ALTERTABLE-NOTES):
    When a column is added with ADD COLUMN and a non-volatile DEFAULT is specified, the default is evaluated at the time of the statement and the result stored in the table's metadata. That value will be used for the column for all existing rows. If no DEFAULT is specified, NULL is used. In neither case is a rewrite of the table required.
So as long as DEFAULT is a static value or a non-volatile expression the table will not need to be rewritten.

Reply:
oh sorry I misread it, you are right. Still wanted to ask if anyone sees an alternative solution so that we don't need to change schema, store more data per event and migrate that data as well